packages/nodejs/src/transmitter.ts

```diff
@@ -0,0 +1,78 @@
+import fs from "fs"
+import path from "path"
+import https from "https"
+import http from "http"
+
+import { Configuration } from "./config"
+import { URL, URLSearchParams } from "url"
+import { HashMap } from "@appsignal/types"
+
+export class Transmitter {
+  #config: Configuration
+  #url: string
+  #data: string
+
+  constructor(url: string, data = "", active = true) {
+    this.#config = new Configuration({ active })
+    this.#url = url
+    this.#data = data
+  }
+
+  public async transmit() {
+    return new Promise<HashMap<any>>((resolve, reject) => {
+      const config_data = this.#config.data
+      const params = new URLSearchParams({
+        api_key: config_data["pushApiKey"] || "",
+        name: config_data["name"] || "",
+        environment: config_data["environment"] || "",
+        hostname: config_data["hostname"] || ""
+      })
+
+      const url = new URL(`${this.#url}?${params.toString()}`)
+      const opts = this.requestOptions(url)
+      const requestModule = url.protocol == "http:" ? http : https
+
+      const request = requestModule.request(opts, (response: any) => {
+        const responseStatus = response.statusCode
+        response.setEncoding("utf8")
+        let responseBody = {}
+
+        response
+          .on("data", (responseData: any) => {
```

Comment:
Is the response data received in the `"data"` event complete? I can't find any authoritative source on this, because bafflingly, the Node.js docs don't seem to document the `"data"` event for [the `http.IncomingRequest` object](https://nodejs.org/api/http.html#class-httpincomingmessage) at all, or at least I can't find where it is documented. But looking at examples sprinkled on the page, it seems like the argument passed to the `"data"` event is a partial chunk of the response, and you're responsible for collecting those chunks into an array or buffer or a string and piecing them together. Quoting [one of the examples sprinkled on the docs](https://nodejs.org/api/http.html#httpgeturl-options-callback):

```js
http.get('http://localhost:8000/', (res) => {
  // [... some stuff we don't care about ...]
  let rawData = '';
  res.on('data', (chunk) => { rawData += chunk; });
  // [...]
}
```

Of course, it doesn't help understand what's going on that many of the examples [[1]](https://nodejs.dev/learn/make-an-http-post-request-using-nodejs) [[2]](https://nodejs.dev/learn/making-http-requests-with-nodejs) [[3]](https://nodejs.org/api/http.html#event-connect) just take the data _chunk_ and push it to standard output, without mentioning that it's a _chunk_ of data, not the full thing, and doing anything with it by itself won't be useful for most usecases. 

Anyway, I think you need to collect the chunks every time the `"data"` event is called, then attempt to `JSON.parse` it on the `"end"` event. Such fun.

Reply:
Sorted out in the latest changes. Now the response is treated as a stream.